add `api/` routes + `newDeck/` page

nelson-lojo opened the pull request:
Add an `/api/chat` route that basically exposes an openai-style chat endpoint for end use without an api key (while the app is running)

SybelBlue commented on src/routes/api/chat/+server.ts:
```diff
@@ -0,0 +1,64 @@
+import { CHAT_API_KEY } from '$env/static/private';
```
Would you mind adding a comment with the file path for where this key should go?

nelson-lojo replied:
Yup, also for future reference it's defined [here](https://kit.svelte.dev/docs/modules#$env-static-private)

SybelBlue reviewed (changes requested):
Looks like a good first step!

nelson-lojo commented:
Summary of changes:
* Added `/api/chat` route that just blankly wraps the openai chat endpoint with only chat functionality
* Added `/api/object` route that allows for instantiating objects and added support for Deck and Card objects as defined by schemas and "type definitions" in `types.d.ts`
* Added a Super Basic deck generation page at `/newDeck` on the site, which POSTs `/api/object` for a Deck and redirects to `/`
* Added an optional url parameter to `/` that allows for custom decks as base64-encoded json

SybelBlue commented on src/lib/types.d.ts:
```diff
@@ -1,2 +1,106 @@
+  }>
+}
+`.trim()
+};
```
This will definitely do for now, but having typedef string literals feels bad. We should chat about this soon.

nelson-lojo replied:
discussed IRL, seems like code expansion is impossible/too esoteric for now

SybelBlue commented on src/routes/newDeck/+page.svelte:
```diff
@@ -0,0 +1,48 @@
+              (deck) => {
+                console.log(deck);
+                let b64payload = btoa(JSON.stringify(deck));
+                goto(`/?customDeckInfo=${b64payload}`);
```
We may want to revise this later - there's gotta be a simpler way of passing data between routes...

SybelBlue commented on src/routes/api/chat/+server.ts:
```diff
@@ -0,0 +1,65 @@
+      /* Potentially useful arguments later on: (pulled from https://platform.openai.com/docs/api-reference/chat/object)
+      n: 1, // how many completions to generate (i.e. how many "attempts" should it make)
+      max_completion_tokens: null, // how many (max) tokens to generate in total
+      response_format: null, // output format: { "type": "json_schema", "json_schema": {...} } ensures the model will match "json_schema" ;;; { "type" : "json_object" } ensures the model will produce valid json
```
Is there a reason we aren't using this for the schema and instead printing typedefs in the prompt?

SybelBlue reviewed (approved):
Looks awesome! Thanks for all the hard work today!!

SybelBlue commented on src/routes/+page.svelte:
```diff
@@ -34,6 +35,25 @@
     })),
   };
 
+  let params = $page.url.searchParams;
```
does this need to be reactive? I'm not sure how query string changes are propagated...
